packages/slice-machine/src/modules/availableCustomTypes/index.ts

```diff
@@ -226,6 +256,49 @@ export function* renameCustomTypeSaga({
   }
 }
 
+export function* deleteCustomTypeSaga({
+  payload,
+}: ReturnType<typeof deleteCustomTypeCreator.request>) {
+  try {
+    yield call(deleteCustomType, payload.customTypeId);
+    yield put(deleteCustomTypeCreator.success(payload));
+    yield put(
+      modalCloseCreator({ modalKey: ModalKeysEnum.DELETE_CUSTOM_TYPE })
+    );
+    yield put(
+      openToasterCreator({
+        message: `Successfully deleted Custom Type “${payload.customTypeName}”`,
+        type: ToasterType.SUCCESS,
+      })
+    );
+  } catch (e) {
+    yield put(
+      modalCloseCreator({ modalKey: ModalKeysEnum.DELETE_CUSTOM_TYPE })
+    );
+    if (axios.isAxiosError(e)) {
+      const apiResponse = e.response?.data as DeleteCustomTypeResponse;
+      console.log(apiResponse);
+      yield put(
+        openToasterCreator({
+          message: apiResponse.reason,
+          type:
+            apiResponse.type === "error"
+              ? ToasterType.ERROR
+              : ToasterType.WARNING,
+        })
+      );
+    } else {
+      yield put(
+        openToasterCreator({
+          message:
+            "An unexpected error happened while deleting your custom type.",
+          type: ToasterType.ERROR,
+        })
+      );
+    }
```

Comment:
This wasn't specified in the ticket, but it's only for the case where the error caught isn't an Axios one. Should never happen but was just to be safe.